src/main/java/org/dependencytrack/model/Repository.java

```diff
@@ -86,6 +86,12 @@ public class Repository implements Serializable {
     @NotNull
     private Boolean internal; // New column, must allow nulls on existing databases
 
+    //New column to determine if authentication is required for a repository
+    @Persistent
+    @Column(name = "AUTHENTICATIONREQUIRED")
+    @NotNull
```

Comment:
As this is a new column, it should be nullable to accommodate for existing repositories that have no value set.

```suggestion
    @Column(name = "AUTHENTICATIONREQUIRED", allowsNull = "true")
```

Alternatively, we'll need an [upgrade](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade) that creates and populates this column for existing repositories.